pkg/agent/multicast/mcast_route.go

```diff
@@ -265,10 +321,10 @@ type multicastInterfaceConfig struct {
 type RouteInterface interface {
 	// MulticastInterfaceJoinMgroup enables interface with name ifaceName and IP ifaceIP
 	// joins multicast group IP mgroup.
```

Comment:
From https://man7.org/linux/man-pages/man7/ip.7.html

  >            The ip_mreqn structure is available only since Linux 2.2.
  >            For compatibility, the old ip_mreq structure (present
  >            since Linux 1.2) is still supported; it differs from
  >            ip_mreqn only by not including the imr_ifindex field.
  >            (The kernel determines which structure is being passed
  >            based on the size passed in optlen.)

Both `ip_mreqn` and `ip_mreq` work for join/leave multicast group as struct passed to `setsockopt`. However, In IPv6 multicast socket programming, only struct `ipv6_mreq` can be accepted. To make it cleaner and more concise in code, I decided to use `ip_mreqn` and `ipv6_mreq` because both structs use ifIndex to reference an interface.

```c
struct ip_mreq  {
	struct in_addr imr_multiaddr;	/* IP multicast address of group */
	struct in_addr imr_interface;	/* local IP address of interface */
};

struct ip_mreqn {
	struct in_addr	imr_multiaddr;		/* IP multicast address of group */
	struct in_addr	imr_address;		/* local IP address of interface */
	int		imr_ifindex;		/* Interface index */
};

struct ipv6_mreq {
	/* IPv6 multicast address of group */
	struct in6_addr ipv6mr_multiaddr;

	/* local IPv6 address of interface */
	int		ipv6mr_ifindex;
};
```